app/src/main/java/com/example/drawingactivity/DrawableAreaView.kt

```diff
@@ -0,0 +1,65 @@
+package com.example.drawingactivity
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.*
+import android.util.AttributeSet
+import android.view.MotionEvent
+import android.view.View
+
+/**
+ * This view is the drawable area of the app.
+ * drawing on this location will draw to a bitmap which is loaded
+ * this saves the paths that are drawn on it
+ */
+@SuppressLint("ClickableViewAccessibility")
+class DrawableAreaView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0
+) : View(context, attrs, defStyle) {
+
+    private var userDrawnBitmap: Bitmap? = null
+    private var intermediaryCanvas: Canvas? = null
```

Comment:
I'm not sure if you need to save this. You can just make one each time you need it

Reply:
This is true, I inlined where it was being assigned a value in the onTouchListener since I never access it outside of that block